src/providers/drafts.ts

```diff
@@ -0,0 +1,122 @@
+  getDraft: (draftId: string) => Draft | undefined
+  getDraftsForChannel: (draftIds: string) => Draft[]
+  getUnseenDraftsForChannel: (draftIds: string) => Draft[]
+  markAllDraftsAsSeenForChannel: (channelId: string) => void
+  removeDrafts: (draftIds: string[]) => void
+  removeAllDrafts: (channelId: string) => void
+}
+
+export const useDraftStore = createStore<DraftStoreStateV1, DraftStoreActions>(
+  {
+    state: {
+      allDrafts: [], // includes drafts for different channels
+    },
+    actionsFactory: (set, get) => ({
+      addDraft: (draft, explicitId) => {
+        const id = explicitId ?? createId()
+        const updatedAt = new Date().toISOString()
+        const newDraft: Draft = { ...draft, updatedAt, id, seen: false }
+        set((draftState) => {
+          draftState.allDrafts = [newDraft, ...draftState.allDrafts]
+        })
+        return newDraft
+      },
+      updateDraft: (draftId, draftProps) => {
+        const updatedAt = new Date().toISOString()
+        set((draftState) => {
+          const idx = draftState.allDrafts.findIndex((d) => d.id === draftId)
+          if (idx >= 0) {
+            draftState.allDrafts[idx] = { ...draftState.allDrafts[idx], ...draftProps, updatedAt }
+          }
+        })
+      },
+      getDraft: (id) => get().allDrafts.find((draft) => draft.id === id),
+      getDraftsForChannel: (channelId) => get().allDrafts.filter((d) => d.channelId === channelId),
+      getUnseenDraftsForChannel: (channelId) =>
+        get().allDrafts.filter((d) => d.channelId === channelId && d.seen === false),
+      removeDrafts: (draftIds) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.filter((draft) => !draftIds.includes(draft.id))
+        })
+      },
+      removeAllDrafts: (channelId) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.filter((draft) => draft.channelId !== channelId)
+        })
+      },
+      markAllDraftsAsSeenForChannel: (channelId) => {
+        set((draftState) => {
+          draftState.allDrafts = draftState.allDrafts.map((draft) => ({
+            ...draft,
+            seen: draft.channelId === channelId ? true : draft.seen,
+          }))
+        })
+      },
+    }),
+  },
+  {
+    persist: {
+      key: 'drafts',
+      whitelist: ['allDrafts'],
+      version: 1,
+      migrate: (oldState, oldVersion, storageValue) => {
+        // migrate store before zustand was added
+        if (oldVersion === undefined) {
+          const unseenDrafts = readFromLocalStorage<
+            Array<{
+              draftId: string
+              channelId: string
+            }>
+          >('unseenDrafts')
+          const drafts = [...(storageValue as Array<Draft>)].map((draft) => {
+            unseenDrafts?.find((unseen) => unseen.draftId === draft.id)
+              ? { ...draft, seen: false }
+              : { ...draft, seen: true }
+          })
```

Comment:
migrate doesn't work great for me. After I switched to your branch I got: 
![image](https://user-images.githubusercontent.com/51168865/123789904-f7b2e300-d8dd-11eb-8d33-029064141126.png) 
And my drafts looks like this: 
![image](https://user-images.githubusercontent.com/51168865/123790025-24ff9100-d8de-11eb-9380-65016569ccf6.png)

Reply:
hmm I will need to check this with someone that still has the old data

Reply:
I guess you could just switch to `dev`, create several drafts and check it yourself ;). Just make sure to copy new drafts somewhere, to reuse them for testing.